### PR TITLE
✨ (feat): Re-add partial upload

### PR DESCRIPTION
**Pull request by WcaleNieWolny**
This commit readds partial upload.
It uses a cloudflare endpoint instead of a supabase one It sends the pre-signed URL request in batches of 10 making it less efficient than before but more stable A new 'cloudflareBaseUrl' option has been added to the capacitor config. This option allows to control the baseUrl for partial upload ZOD validation has beed added compared to old partial upload. This decreases the chance of bugs occurring

### CODE REVIEW

File: src/bundle/partial.ts [hunk 1]
```diff
@@ -27,49 +43,65 @@ export async function prepareBundlePartialFiles(path: string, snag: LogSnag, org
...
 }
 
-export async function uploadPartial(apikey: string, manifest: manifestType, path: string, options: Options, appId: string, name: string) {
-  const spinner = spinnerC()
-  spinner.start('Preparing partial update')
-  const uploadResponse: uploadUrlsType[] = await manifestUploadUrls(apikey, appId, name, manifest)
+export async function uploadPartial(apikey: string, manifest: manifestType, path: string, options: any, config: CapacitorConfig, appId: string, name: string) {
+  try {
+    const baseUrl = config?.plugins?.CapacitorUpdater?.cloudflareBaseUrl ?? defaultApiHost
+    const url = new URL(`${baseUrl}/private/partial_upload/v1`)
```
Comment (riderx) [inline src/bundle/partial.ts:49]: Do not put v1 in the URL, we will do that in the headers later

File: src/bundle/partial.ts [hunk 2]
```diff
@@ -27,49 +43,65 @@ export async function prepareBundlePartialFiles(path: string, snag: LogSnag, org
...
 
-  if (uploadResponse.length === 0 || uploadResponse.length !== manifest.length) {
-    log.error(`Cannot upload manifest, please try again later`)
-    spinner.stop('Partial update failed')
-    return []
-  }
-  spinner.message('Uploading partial update')
-  for (const [index, manifestEntry] of uploadResponse.entries()) {
-    const finalFilePath = `${path}/${manifestEntry.path}`
-    spinner.message(`Uploading partial update ${index + 1}/${uploadResponse.length}`)
```
Comment (riderx) [inline src/bundle/partial.ts:43]: add back the spinner

File: src/bundle/partial.ts [hunk 3]
```diff
@@ -27,49 +43,65 @@ export async function prepareBundlePartialFiles(path: string, snag: LogSnag, org
   return manifest
 }
 
-export async function uploadPartial(apikey: string, manifest: manifestType, path: string, options: Options, appId: string, name: string) {
-  const spinner = spinnerC()
-  spinner.start('Preparing partial update')
-  const uploadResponse: uploadUrlsType[] = await manifestUploadUrls(apikey, appId, name, manifest)
+export async function uploadPartial(apikey: string, manifest: manifestType, path: string, options: any, config: CapacitorConfig, appId: string, name: string) {
+  try {
+    const baseUrl = config?.plugins?.CapacitorUpdater?.cloudflareBaseUrl ?? defaultApiHost
```
Comment (riderx) [inline src/bundle/partial.ts:48]: Rename cloudflareBaseUrl to something more generic like partialUploadUrl, and more specific to the endpoint

File: src/bundle/partial.ts (hunk 1 above)
Comment (WcaleNieWolny) [reply src/bundle/partial.ts:49]: You cannot route in hono based on headers. It doesn't hurt to keep the v1

File: src/bundle/partial.ts (hunk 2 above)
Comment (WcaleNieWolny) [reply src/bundle/partial.ts:43]: There is a spinner, just look in the upload code directly

**Comment by riderx** [comment]
close in fav of TOS

**Pull request closed without merging**